components/dot_menu/dot_menu.tsx

```diff
@@ -338,6 +339,32 @@ export class DotMenuClass extends React.PureComponent<Props, State> {
         }
     }
 
+    renderSubMenus = (bindings: AppBinding[] = []): SubMenuProps[] => {
+        return bindings.map((b) => {
+            let icon: React.ReactNode;
+            if (b.icon) {
+                icon = (<img src={b.icon}/>);
+            }
+            let action: ((id?: string | undefined) => void) | undefined;
+            let subMenu: SubMenuProps[] | undefined;
+
+            if (b.bindings?.length) {
+                subMenu = this.renderSubMenus(b.bindings);
+            } else {
+                action = () => this.onClickAppBinding(b);
+            }
```

Comment:
Should we limit how many levels you can render the sub menus?

Reply:
This is something that needs further feedback from UX.

Among other things, there has been talks to get all plugable actions to be in a submenu, and do not allow further submenus.